### PR TITLE
ANSI-752 - First class basic auth support

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Greatly simplifies configuring basic auth for our components. Now you have `<component>_authentication_type: basic`
Also deprecates `<component>_ssl_mutual_auth_enabled` var in favor of setting `<component>_authentication_type: mtls`

Docs:
RP: https://docs.confluent.io/platform/current/kafka-rest/production-deployment/rest-proxy/security.html#http-basic-authentication
Connect: https://docs.confluent.io/platform/current/security/basic-auth.html#crest
SR: https://docs.confluent.io/platform/current/security/basic-auth.html#schema-registry
ksqlDB: https://docs.confluent.io/platform/current/security/basic-auth.html#ksqldb
C3: https://docs.confluent.io/platform/current/security/basic-auth.html#c3-short-rest-api

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

updates the `plaintext-basic-rhel` scenario

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.test/molecule/plaintext-basic-rhel/molecule.yml [hunk 1]
```diff
@@ -124,92 +124,101 @@ provisioner:
...
+        # C3 Basic Auth
+        control_center_authentication_type: basic
+        # control_center_copy_files:
+        #   - source_path: "roles/confluent.test/molecule/{{scenario_name}}/password.properties"
+        #     destination_path: /etc/kafka-rest/password.properties
+        #   - source_path: "roles/confluent.test/molecule/{{scenario_name}}/rest-jaas.properties"
+        #     destination_path: /etc/kafka-rest/rest-jaas.properties
+
+        # control_center_custom_properties:
+        #   confluent.controlcenter.rest.authentication.method: BASIC
```
Comment (domenicbove) [inline roles/confluent.test/molecule/plaintext-basic-rhel/molecule.yml:212]: remove commented out code

File: roles/confluent.variables/vars/main.yml [hunk 2]
```diff
@@ -273,6 +278,14 @@ kafka_broker_properties:
...
       confluent.http.server.ssl.keystore.type: BCFKS
       confluent.http.server.ssl.truststore.type: BCFKS
+  embedded_rest_proxy_basic:
+    enabled: "{{ kafka_broker_rest_proxy_enabled and (not rbac_enabled or (rbac_enabled and external_mds_enabled)) and kafka_broker_rest_proxy_authentication_type == 'basic' }}"
+    properties:
+      kafka.rest.resource.extension.class: io.confluent.kafkarest.security.KafkaRestSecurityResourceExtension
+      kafka.rest.confluent.rest.auth.propogate.method: JETTY_AUTH
+      kafka.rest.authentication.method: BASIC
+      kafka.rest.authentication.realm: KafkaRest
+      kafka.rest.authentication.roles: "{{kafka_broker_rest_proxy_basic_users.admin.roles}}"
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:288]: A loop here would be superior. ie loop over all subdicts, combine their roles

File: roles/confluent.test/molecule/plaintext-basic-rhel/verify.yml [hunk 3]
```diff
@@ -1,16 +1,98 @@
...
+        file_path: /etc/kafka/connect-distributed.properties
+        property: rest.extension.classes
+        expected_value: org.apache.kafka.connect.rest.basic.auth.extension.BasicAuthSecurityRestExtension
+
+    - name: Load connect-password.properties
+      slurp:
+        src: /etc/kafka/connect-password.properties
+      register: slurped_properties
+
+    - debug:
```
Comment (domenicbove) [inline roles/confluent.test/molecule/plaintext-basic-rhel/verify.yml:55]: dont need this debug

File: roles/confluent.variables/vars/main.yml [hunk 4]
```diff
@@ -436,9 +449,16 @@ schema_registry_properties:
...
-    enabled: "{{ schema_registry_ssl_mutual_auth_enabled }}"
+    enabled: "{{ schema_registry_authentication_type == 'mtls' }}"
     properties:
       ssl.client.auth: 'true'
+  basic:
+    enabled: "{{ schema_registry_authentication_type == 'basic' }}"
+    properties:
+      authentication.method: BASIC
+      authentication.realm: SchemaRegistry
+      # authentication.roles: "{{schema_registry_basic_users.admin.roles}}"
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:460]: remove comment

File: roles/confluent.variables/vars/main.yml [hunk 5]
```diff
@@ -751,9 +779,16 @@ ksql_properties:
...
-    enabled: "{{ ksql_ssl_mutual_auth_enabled }}"
+    enabled: "{{ ksql_authentication_type == 'mtls' }}"
     properties:
       ssl.client.auth: 'true'
+  basic:
+    enabled: "{{ ksql_authentication_type == 'basic' }}"
+    properties:
+      authentication.method: BASIC
+      authentication.realm: KsqlServer
+      # authentication.roles: "{{ksql_basic_users.admin.roles}}"
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:790]: remove comment

File: roles/confluent.variables/vars/main.yml [hunk 6]
```diff
@@ -911,9 +951,16 @@ kafka_rest_properties:
...
-    enabled: "{{ kafka_rest_ssl_mutual_auth_enabled }}"
+    enabled: "{{ kafka_rest_authentication_type == 'mtls' }}"
     properties:
       ssl.client.auth: 'true'
+  basic:
+    enabled: "{{ kafka_rest_authentication_type == 'basic' }}"
+    properties:
+      authentication.method: BASIC
+      authentication.realm: KafkaRest
+      # authentication.roles: "{{kafka_rest_basic_users.admin.roles}}"
```
Comment (domenicbove) [inline roles/confluent.variables/vars/main.yml:962]: remove comment

**Review by xnulinu — APPROVED** [review]

**Review by lokesh-shekar — APPROVED** [review]
LGTM. Thanks @domenicbove 👍

**Pull request merged**